com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -475,9 +475,7 @@ public virtual void InitializeAgent()
         /// </returns>
         public virtual float[] Heuristic()
         {
-            throw new UnityAgentsException(
-                "The Heuristic method was not implemented for the Agent on the " +
-                $"{gameObject.name} GameObject.");
+            return null;
```

Comment:
If we want to keep this usable without inheriting, I think we need to remove this exception. Will to be overruled though.

Reply:
Why do we need to remove it? If Heuristic is called but not implemented, an error should be raised. Or should be make a Heuristic component?
Other question, is it okay for it to be null? Does that break something down the line?

Reply:
How about a log warning instead of an exception?

> is it okay for it to be null?

I think that would depend on the AgentAction implementation.